server/src/test/java/org/elasticsearch/index/translog/TestTranslog.java

```diff
@@ -53,6 +55,31 @@
 public class TestTranslog {
     static final Pattern TRANSLOG_FILE_PATTERN = Pattern.compile("translog-(\\d+)\\.tlog");
 
+    /*TODO currently there are two methods for corrupting translog.
+    There are two differences:
+     1) corruptTranslogFiles method is expecting other index files, not only translog.
+     2) corruptTranslogFiles method sometimes does not corrupt translog at all.
+    //It would be nice to have single corruption method to be used in integration and unit tests*/
+    /**
+     * Randomly overwrite some bytes in the translog files
+     */
+    public static void corruptTranslogFilesReliably(Logger logger, Random random, Path directory) throws Exception {
+        Path[] files = FileSystemUtils.files(directory, "translog-*");
+        for (Path file : files) {
+            logger.info("--> corrupting {}...", file);
+            FileChannel f = FileChannel.open(file, StandardOpenOption.READ, StandardOpenOption.WRITE);
+            int corruptions = RandomizedTest.scaledRandomIntBetween(10, 50);
+            for (int i = 0; i < corruptions; i++) {
+                // note: with the current logic, this will sometimes be a no-op
+                long pos = RandomNumbers.randomIntBetween(random,0, (int) f.size());
+
+                ByteBuffer junk = ByteBuffer.wrap(new byte[]{(byte) random.nextInt()});
```

Comment:
In theory this could overwrite a byte with the same byte, which isn't a corruption. But then it overwrites at least 10 bytes in this file, so the chances are it won't do this 10 times. But then overwriting 10-50 random bytes is not a very realistic corruption: I'd expect something small (a single bit flip) or else something enormous (whole sectors are zeroed or contain the wrong data). I prefer the other method.